digdag-standards/src/main/java/io/digdag/standards/command/ecs/EcsClientConfig.java

```diff
@@ -47,40 +82,36 @@ private static EcsClientConfig createFromSystemConfig(final Optional<String> clu
 
         final String extractedPrefix = SYSTEM_CONFIG_PREFIX + name + ".";
         final Config extracted = StorageManager.extractKeyPrefix(systemConfig, extractedPrefix);
-        return new EcsClientConfig(name,
-                extracted.get("launch_type", String.class),
-                extracted.get("access_key_id", String.class),
-                extracted.get("secret_access_key", String.class),
-                extracted.get("region", String.class),
-                extracted.get("subnets", String.class),
-                extracted.get("max_retries", int.class, 3)
-        );
+
+        return buildEcsClientConfig(name, extracted);
+    }
+
+    private static EcsClientConfig buildEcsClientConfig(String clusterName, Config ecsConfig)
+    {
+        return EcsClientConfig.builder()
+                .withClusterName(clusterName)
+                .withLaunchType(ecsConfig.get("launch_type", String.class))
+                .withAccessKeyId(ecsConfig.get("access_key_id", String.class))
+                .withSecretAccessKey(ecsConfig.get("secret_access_key", String.class))
+                .withRegion(ecsConfig.get("region", String.class))
+                .withSubnets(ecsConfig.getOptional("subnets", String.class))
+                .withMaxRetries(ecsConfig.get("max_retries", int.class, DEFAULT_MAX_TRIES))
+                .withCapacityProviderName(ecsConfig.getOptional("capacity_provider_name", String.class))
+                .withCpu(ecsConfig.getOptional("cpu", Integer.class))
+                .withCpu(ecsConfig.getOptional("memory", Integer.class))
+                .build();
     }
 
     private final String clusterName;
     private final String launchType;
     private final String accessKeyId;
     private final String secretAccessKey;
     private final String region;
-    private final List<String> subnets;
+    private final Optional<List<String>> subnets;
     private final int maxRetries;
-
-    private EcsClientConfig(final String clusterName,
-            final String launchType,
-            final String accessKeyId,
-            final String secretAccessKey,
-            final String region,
-            final String subnets,
-            final int maxRetries)
-    {
-        this.clusterName = clusterName;
-        this.launchType = launchType;
-        this.accessKeyId = accessKeyId;
-        this.secretAccessKey = secretAccessKey;
-        this.region = region;
-        this.subnets = Arrays.asList(subnets.split(",")); // TODO more robust
-        this.maxRetries = maxRetries;
-    }
```

Comment:
I deleted current Constructor because this Constructor has a lot of arguments and in the future the number of this parameter is considered to increase.